internal/serviceconfig/serviceconfig_test.go

```diff
@@ -0,0 +1,125 @@
+/*
+ *
+ * Copyright 2020 gRPC authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package serviceconfig
+
+import (
+	"encoding/json"
+	"fmt"
+	"testing"
+
+	"github.com/google/go-cmp/cmp"
+	"google.golang.org/grpc/balancer"
+	"google.golang.org/grpc/balancer/roundrobin"
+	externalserviceconfig "google.golang.org/grpc/serviceconfig"
+)
+
+type testBalancerConfigType struct {
+	externalserviceconfig.LoadBalancingConfig
+}
+
+var testBalancerConfig = testBalancerConfigType{}
+
+const (
+	testBalancerBuilderName = "test-bb"
+	testBalancerConfigJson  = `{"test-balancer-config":"true"}`
+)
+
+type testBalancerBuilder struct{}
+
+func (testBalancerBuilder) ParseConfig(js json.RawMessage) (externalserviceconfig.LoadBalancingConfig, error) {
+	if string(js) != testBalancerConfigJson {
+		return nil, fmt.Errorf("unexpected config json")
+	}
+	return testBalancerConfig, nil
+}
+
+func (testBalancerBuilder) Build(cc balancer.ClientConn, opts balancer.BuildOptions) balancer.Balancer {
+	panic("implement me")
+}
+
+func (testBalancerBuilder) Name() string {
+	return testBalancerBuilderName
+}
+
+func init() {
+	balancer.Register(testBalancerBuilder{})
+}
+
+func TestBalancerConfigUnmarshalJSON(t *testing.T) {
+	tests := []struct {
+		name    string
+		json    string
+		want    BalancerConfig
+		wantErr bool
+	}{
+		{
+			name:    "empty json",
+			json:    "",
+			wantErr: true,
+		},
+		{
+			name:    "more than one entry for a name",
+			json:    `[{"balancer1":"1","balancer2":"2"}]`,
```

Comment:
should these both be "balancer1" then?

Reply:
This is a map with two entries, we want one entry.
Changed name and added comment.